From 2a1a20ae5273ead9833ffabade5fb737050e47a7 Mon Sep 17 00:00:00 2001
From: Han Wang <416810799@qq.com>
Date: Sun, 30 Sep 2018 10:57:48 +0200
Subject: [PATCH 2/2] DocumentsUI: Don't check for mStarted state on
 user/userdebug builds

 * I can reproduce this crash 100%: Open DocumentsUI, long select an
   download file, then press back button to cancel it. Ater redoing this
   for 3-4 times DocumentsUI will crash.
 * Stacktrace shows that it crashes at checkState(!mStarted) but I don't
   know the root cause. There is a similiar check several lines below
   which only allows the check on user builds. Since we ship userdebug
   builds, make it check for eng build only.

Change-Id: Ic6c24b4eb9fb5f1e65c0a37c6488b5d0acff8b0b
---
 .../android/documentsui/selection/GestureSelectionHelper.java | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/com/android/documentsui/selection/GestureSelectionHelper.java b/src/com/android/documentsui/selection/GestureSelectionHelper.java
index d3d8be16..410c7530 100644
--- a/src/com/android/documentsui/selection/GestureSelectionHelper.java
+++ b/src/com/android/documentsui/selection/GestureSelectionHelper.java
@@ -81,7 +81,7 @@ public final class GestureSelectionHelper extends ScrollHost implements OnItemTo
      * @return true if started.
      */
     public void start() {
-        checkState(!mStarted);
+        if (Build.IS_ENG) checkState(!mStarted);
         // See: b/70518185. It appears start() is being called via onLongPress
         // even though we never received an intial handleInterceptedDownEvent
         // where we would usually initialize mLastStartedItemPos.
@@ -128,7 +128,7 @@ public final class GestureSelectionHelper extends ScrollHost implements OnItemTo
         // after combinations of mouse + touch + rotation.
         // But after further investigation I couldn't repro.
         // For that reason we guard this check (for now) w/ IS_DEBUGGABLE.
-        if (Build.IS_DEBUGGABLE) checkState(mStarted);
+        if (Build.IS_ENG) checkState(mStarted);
 
         switch (e.getActionMasked()) {
             case MotionEvent.ACTION_MOVE:
-- 
2.17.1

